src/mmrelay/config.py

```
@@ -815,147 +893,67 @@ def save_credentials(
+    try:
+        os.makedirs(target_dir, exist_ok=True)
     except (OSError, PermissionError):
+        logger.exception("Could not create credentials directory %s", target_dir)
         if sys.platform == "win32":
-            logger.exception(
-                "Error saving credentials.json to %s. On Windows, ensure the application "
-                "has write permissions to the user data directory.",
-                config_dir,
+            logger.error(
+                "On Windows, ensure the application has write permissions to the credentials path."
             )
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)